### ISSUE TITLE
Crash on trying to register

### ISSUE THREAD

**Issue by danialbehzadi** [UP-NextPush/android#44]
I set up Unified push on my Nextcloud and logged in with my Nexcloud account in NextPush. But every time I try to register with Up-Example, NextPush crahes.

here is a logcat:
```
25380 06-22 12:36:04.719 19131 19131 D UnifiedPush: Found distributor with package name org.unifiedpush.distributor.nextpush
25381 06-22 12:36:04.719 19131 19131 D UnifiedPush: Found saved distributor.
25382 06-22 12:36:04.726 28065 28065 I RegisterBroadcastReceiver: REGISTER
25383 06-22 12:36:04.737 28065 28065 D AndroidRuntime: Shutting down VM
25384 06-22 12:36:04.738 28065 28065 E AndroidRuntime: FATAL EXCEPTION: main
25385 06-22 12:36:04.738 28065 28065 E AndroidRuntime: Process: org.unifiedpush.distributor.nextpush, PID: 28065
25386 06-22 12:36:04.738 28065 28065 E AndroidRuntime: java.lang.RuntimeException: Unable to start receiver org.unifiedpush.distributor.nextpu      sh.receivers.RegisterBroadcastReceiver: java.lang.NullPointerException
25387 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at android.app.ActivityThread.handleReceiver(ActivityThread.java:4384)
25388 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at android.app.ActivityThread.access$1600(ActivityThread.java:256)
25389 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2102)
25390 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:106)
25391 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at android.os.Looper.loopOnce(Looper.java:201)
25392 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:288)
25393 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:7870)
25394 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
25395 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:      548)
25396 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
25397 06-22 12:36:04.738 28065 28065 E AndroidRuntime: Caused by: java.lang.NullPointerException
25398 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at org.unifiedpush.distributor.nextpush.api.ApiUtils.cCreateApp(ApiUtils.kt:188)
25399 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at org.unifiedpush.distributor.nextpush.api.ApiUtils.access$cCreateApp(ApiUtils.      kt:30)
25400 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at org.unifiedpush.distributor.nextpush.api.ApiUtils$apiCreateApp$1.invoke(ApiUt      ils.kt:174)
25401 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at org.unifiedpush.distributor.nextpush.api.ApiUtils$apiCreateApp$1.invoke(ApiUt      ils.kt:173)
25402 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at org.unifiedpush.distributor.nextpush.api.ApiUtils.cApi(ApiUtils.kt:49)
25403 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at org.unifiedpush.distributor.nextpush.api.ApiUtils.apiCreateApp(ApiUtils.kt:17      3)
25404 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at org.unifiedpush.distributor.nextpush.receivers.RegisterBroadcastReceiver.onRe      ceive(RegisterBroadcastReceiver.kt:71)
25405 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        at android.app.ActivityThread.handleReceiver(ActivityThread.java:4375)
25406 06-22 12:36:04.738 28065 28065 E AndroidRuntime:        ... 9 more
```

**Comment by p1gp1g** [comment]
Is there any error message at https://your.nc.domain/settings/admin/uppush ?

**Comment by Bubu** [comment]
I also got this I think, here's a more complete crash log:
```
--------- beginning of crash
07-05 00:19:24.144  2125  2125 E AndroidRuntime: FATAL EXCEPTION: main
07-05 00:19:24.144  2125  2125 E AndroidRuntime: Process: org.unifiedpush.distributor.nextpush, PID: 2125
07-05 00:19:24.144  2125  2125 E AndroidRuntime: java.lang.RuntimeException: Unable to start receiver org.unifiedpush.distributor.nextpush.receivers.RegisterBroadcastReceiver: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.google.gson.Gson.toJson(java.lang.Object)' on a null object reference
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at android.app.ActivityThread.handleReceiver(ActivityThread.java:4384)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at android.app.ActivityThread.access$1600(ActivityThread.java:256)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2102)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7870)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.google.gson.Gson.toJson(java.lang.Object)' on a null object reference
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at com.nextcloud.android.sso.api.NextcloudRetrofitServiceMethod.invoke(NextcloudRetrofitServiceMethod.java:133)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at retrofit2.NextcloudRetrofitApiBuilder.lambda$create$0$retrofit2-NextcloudRetrofitApiBuilder(NextcloudRetrofitApiBuilder.java:30)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at retrofit2.NextcloudRetrofitApiBuilder$$ExternalSyntheticLambda0.invoke(Unknown Source:2)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at java.lang.reflect.Proxy.invoke(Proxy.java:1006)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at $Proxy2.createApp(Unknown Source)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at org.unifiedpush.distributor.nextpush.api.ApiUtils.cCreateApp(ApiUtils.kt:191)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at org.unifiedpush.distributor.nextpush.api.ApiUtils.access$cCreateApp(ApiUtils.kt:30)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at org.unifiedpush.distributor.nextpush.api.ApiUtils$apiCreateApp$1.invoke(ApiUtils.kt:174)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at org.unifiedpush.distributor.nextpush.api.ApiUtils$apiCreateApp$1.invoke(ApiUtils.kt:173)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at org.unifiedpush.distributor.nextpush.api.ApiUtils.cApi(ApiUtils.kt:49)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at org.unifiedpush.distributor.nextpush.api.ApiUtils.apiCreateApp(ApiUtils.kt:173)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at org.unifiedpush.distributor.nextpush.receivers.RegisterBroadcastReceiver.onReceive(RegisterBroadcastReceiver.kt:71)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	at android.app.ActivityThread.handleReceiver(ActivityThread.java:4375)
07-05 00:19:24.144  2125  2125 E AndroidRuntime: 	... 9 more
```

**Comment by danialbehzadi** [comment]
My issue seems to be resolved in 1.0.5.

What about you @Bubu?

**Comment by Bubu** [comment]
I'm just saw this again:

```
09-30 13:27:37.726 23561 23561 E AndroidRuntime: Process: org.unifiedpush.distributor.nextpush, PID: 23561
09-30 13:27:37.726 23561 23561 E AndroidRuntime: java.lang.RuntimeException: Unable to start receiver org.unifiedpush.distributor.nextpush.receivers.RegisterBroadcastReceiver: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.google.gson.Gson.toJson(java.lang.Object)' on a null object reference
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at android.app.ActivityThread.handleReceiver(ActivityThread.java:4345)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleReceiver(Unknown Source:0)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2154)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7898)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.google.gson.Gson.toJson(java.lang.Object)' on a null object reference
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at com.nextcloud.android.sso.api.NextcloudRetrofitServiceMethod.invoke(NextcloudRetrofitServiceMethod.java:133)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at retrofit2.NextcloudRetrofitApiBuilder.lambda$create$0$retrofit2-NextcloudRetrofitApiBuilder(NextcloudRetrofitApiBuilder.java:30)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at retrofit2.NextcloudRetrofitApiBuilder$$ExternalSyntheticLambda0.invoke(Unknown Source:2)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at java.lang.reflect.Proxy.invoke(Proxy.java:1006)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at $Proxy2.createApp(Unknown Source)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at org.unifiedpush.distributor.nextpush.api.ApiUtils.cCreateApp(ApiUtils.kt:198)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at org.unifiedpush.distributor.nextpush.api.ApiUtils.access$cCreateApp(ApiUtils.kt:30)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at org.unifiedpush.distributor.nextpush.api.ApiUtils$apiCreateApp$1.invoke(ApiUtils.kt:178)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at org.unifiedpush.distributor.nextpush.api.ApiUtils$apiCreateApp$1.invoke(ApiUtils.kt:177)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at org.unifiedpush.distributor.nextpush.api.ApiUtils.cApi(ApiUtils.kt:49)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at org.unifiedpush.distributor.nextpush.api.ApiUtils.apiCreateApp(ApiUtils.kt:177)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at org.unifiedpush.distributor.nextpush.receivers.RegisterBroadcastReceiver.onReceive(RegisterBroadcastReceiver.kt:71)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	at android.app.ActivityThread.handleReceiver(ActivityThread.java:4336)
09-30 13:27:37.726 23561 23561 E AndroidRuntime: 	... 9 more
```

This happened after I unregistered everything in nextpush android (because schildichat was there twice and tusky was still there despite being uninstalled). And then trying to select Nextpush again from the schildichat notification system selection (it defaulted to background polling).

**Comment by p1gp1g** [comment]
I've been able to reproduce, I'll investigate this week end

```
09-30 13:36:19.989 20637 20637 I RegisterBroadcastReceiver: UNREGISTER
09-30 13:36:19.990 20637 20637 D com.nextcloud.android.sso.api.NextcloudRetrofitServiceMethod: invoke call to api using observable class org.unifiedpush.distributor.nextpush.api.ApiResponse
09-30 13:36:19.990 20637 20637 D ApiUtils: Subscribed to deleteApp.
09-30 13:36:19.991 20637 20833 V com.nextcloud.android.sso.api.AidlNetworkRequest: [waitForApi] - api not ready yet.. waiting [RxNewThreadScheduler-15]
09-30 13:36:21.735 20637 20637 I RegisterBroadcastReceiver: REGISTER
09-30 13:36:21.736 20637 20637 D AndroidRuntime: Shutting down VM
--------- beginning of crash
09-30 13:36:21.737 20637 20637 E AndroidRuntime: FATAL EXCEPTION: main
09-30 13:36:21.737 20637 20637 E AndroidRuntime: Process: org.unifiedpush.distributor.nextpush, PID: 20637
09-30 13:36:21.737 20637 20637 E AndroidRuntime: java.lang.RuntimeException: Unable to start receiver org.unifiedpush.distributor.nextpush.receivers.RegisterBroadcastReceiver: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.google.gson.Gson.toJson(java.lang.Object)' on a null object reference
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at android.app.ActivityThread.handleReceiver(ActivityThread.java:4346)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at android.app.ActivityThread.-$$Nest$mhandleReceiver(Unknown Source:0)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2155)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:106)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at android.os.Looper.loopOnce(Looper.java:201)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:288)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:7904)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at com.android.internal.os.ExecInit.main(ExecInit.java:49)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:355)
09-30 13:36:21.737 20637 20637 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.google.gson.Gson.toJson(java.lang.Object)' on a null object reference
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at com.nextcloud.android.sso.api.NextcloudRetrofitServiceMethod.invoke(NextcloudRetrofitServiceMethod.java:133)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at retrofit2.NextcloudRetrofitApiBuilder.lambda$create$0$retrofit2-NextcloudRetrofitApiBuilder(NextcloudRetrofitApiBuilder.java:30)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at retrofit2.NextcloudRetrofitApiBuilder$$ExternalSyntheticLambda0.invoke(Unknown Source:2)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at java.lang.reflect.Proxy.invoke(Proxy.java:1006)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at $Proxy1.createApp(Unknown Source)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at org.unifiedpush.distributor.nextpush.api.ApiUtils.cCreateApp(ApiUtils.kt:198)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at org.unifiedpush.distributor.nextpush.api.ApiUtils.access$cCreateApp(ApiUtils.kt:30)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at org.unifiedpush.distributor.nextpush.api.ApiUtils$apiCreateApp$1.invoke(ApiUtils.kt:178)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at org.unifiedpush.distributor.nextpush.api.ApiUtils$apiCreateApp$1.invoke(ApiUtils.kt:177)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at org.unifiedpush.distributor.nextpush.api.ApiUtils.cApi(ApiUtils.kt:49)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at org.unifiedpush.distributor.nextpush.api.ApiUtils.apiCreateApp(ApiUtils.kt:177)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at org.unifiedpush.distributor.nextpush.receivers.RegisterBroadcastReceiver.onReceive(RegisterBroadcastReceiver.kt:71)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        at android.app.ActivityThread.handleReceiver(ActivityThread.java:4337)
09-30 13:36:21.737 20637 20637 E AndroidRuntime:        ... 11 more
09-30 13:36:29.998 20637 20839 W System.err: com.nextcloud.android.sso.exceptions.NextcloudApiNotRespondingException: Pas de réponse de l'API de l'application Nextcloud Files. Cela est probablement dû à des optimalisations de la batterie de votre téléphone. Essayez de désactiver l'optimalisation de la batterie pour les 2 applications (Nextcloud Android et cette application). Sinon, essayez de remettre à zero le stockage de ces 2 applications
09-30 13:36:29.998 20637 20839 W System.err:    at com.nextcloud.android.sso.api.AidlNetworkRequest.waitForApi(AidlNetworkRequest.java:162)
09-30 13:36:29.998 20637 20839 W System.err:    at com.nextcloud.android.sso.api.AidlNetworkRequest.performAidlNetworkRequestV2(AidlNetworkRequest.java:298)
09-30 13:36:29.998 20637 20839 W System.err:    at com.nextcloud.android.sso.api.AidlNetworkRequest.performNetworkRequestV2(AidlNetworkRequest.java:180)
09-30 13:36:29.998 20637 20839 W System.err:    at com.nextcloud.android.sso.api.NextcloudAPI.performNetworkRequestV2(NextcloudAPI.java:199)
09-30 13:36:29.998 20637 20839 W System.err:    at com.nextcloud.android.sso.api.NextcloudAPI.lambda$performRequestObservableV2$1$com-nextcloud-android-sso-api-NextcloudAPI(NextcloudAPI.java:129)
09-30 13:36:29.999 20637 20839 W System.err:    at com.nextcloud.android.sso.api.NextcloudAPI$$ExternalSyntheticLambda0.subscribe(Unknown Source:6)
09-30 13:36:29.999 20637 20839 W System.err:    at io.reactivex.internal.operators.observable.ObservableFromPublisher.subscribeActual(ObservableFromPublisher.java:31)
09-30 13:36:29.999 20637 20839 W System.err:    at io.reactivex.Observable.subscribe(Observable.java:12284)
09-30 13:36:29.999 20637 20839 W System.err:    at io.reactivex.internal.operators.observable.ObservableMap.subscribeActual(ObservableMap.java:32)
09-30 13:36:29.999 20637 20839 W System.err:    at io.reactivex.Observable.subscribe(Observable.java:12284)
09-30 13:36:29.999 20637 20839 W System.err:    at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
09-30 13:36:29.999 20637 20839 W System.err:    at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:608)
09-30 13:36:29.999 20637 20839 W System.err:    at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
09-30 13:36:29.999 20637 20839 W System.err:    at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
09-30 13:36:29.999 20637 20839 W System.err:    at java.util.concurrent.FutureTask.run(FutureTask.java:264)
09-30 13:36:29.999 20637 20839 W System.err:    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:307)
09-30 13:36:29.999 20637 20839 W System.err:    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
09-30 13:36:29.999 20637 20839 W System.err:    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
09-30 13:36:29.999 20637 20839 W System.err:    at java.lang.Thread.run(Thread.java:1012)
```

**Comment by p1gp1g** [comment]
Fixed with 1.4.1
